### PR TITLE
Zhiwen/getting nearby codes

### LINKED ISSUE #21: US 07.01.01
As a player, I want to see game-wide high scores of all players.

### PR DESCRIPTION
**Pull request by real-rookie**

### CODE REVIEW

File: app/src/main/java/com/cmput301w23t40/capturetheqr/ScoreboardActivity.java [hunk 1]
```diff
@@ -39,125 +57,138 @@ public class ScoreboardActivity extends AppCompatActivity implements AdapterView
...
+                                    my_player = tempPlayer;
+                                    foundMyPlayer = true;
+                                }
+                            }
+                            playerAdapter.notifyDataSetChanged();
+                            listView.setAdapter(playerAdapter);
+                            setMyRankText(my_player.getRank(), itemSelected);
+                        } else {
+                            updateLocation();
+                            locationHelper.updateLocation(new PlayerLocation.CallbackLocation() {
```
Comment (nic3-14159) [inline app/src/main/java/com/cmput301w23t40/capturetheqr/ScoreboardActivity.java]: locationHelper will be null here if the user does not grant location permissions, as the only places it is initialized is inside blocks where we have those permissions. The code should handle the case where no location permissions were granted by either not proceeding with the requested action (in AddQRActivity I unset the save geolocation toggle) or allowing a different way to get the location (such as a manual location entry)

File: app/src/main/java/com/cmput301w23t40/capturetheqr/ScoreboardActivity.java [hunk 2]
```diff
@@ -15,21 +16,38 @@
...
+    private String deviceID;
+    private PlayerLocation locationHelper;
 
+    private ActivityResultLauncher<String[]> locationPermissionLauncher = registerForActivityResult(
+            new ActivityResultContracts.RequestMultiplePermissions(), result -> {
+        /* result is a map with the permission name as the key and the value being a
+        boolean indicating whether the permission was granted */
+                // User granted location permissions
+                if (result.get("android.permission.ACCESS_COARSE_LOCATION") &&
+                        result.get("android.permission.ACCESS_COARSE_LOCATION")) {
```
Comment (nic3-14159) [inline app/src/main/java/com/cmput301w23t40/capturetheqr/ScoreboardActivity.java]: We don't need two checks for coarse location permissions. That was an typo in my original code that I had sent you before, but it is now fixed in AddQRActivity which changes the second one checks `ACCESS_FINE_LOCATION`. 

Though, for the purposes of checking a region we probably don't need fine location permissions (accurate to a 10-50 meter distance) and could probably just use coarse location (accurate to within a 3 km^2  area) https://developer.android.com/training/location/permissions#accuracy

**Review by nic3-14159 — CHANGES_REQUESTED** [review]
Some comments about using location services

File: app/src/main/java/com/cmput301w23t40/capturetheqr/ScoreboardActivity.java (hunk 1 above)
Comment (real-rookie) [reply app/src/main/java/com/cmput301w23t40/capturetheqr/ScoreboardActivity.java]: done

File: app/src/main/java/com/cmput301w23t40/capturetheqr/ScoreboardActivity.java (hunk 2 above)
Comment (real-rookie) [reply app/src/main/java/com/cmput301w23t40/capturetheqr/ScoreboardActivity.java]: done

**Comment by real-rookie** [comment]
> Some comments about using location services

Sorry, I just need to have a break now. Will add comments to all the code I wrote together tonight

File: app/src/main/res/values/strings.xml
Comment (rachel-ellis) [inline app/src/main/res/values/strings.xml:1]: I think you accidentally undid @ekhumbata's text changes here

**Review by rachel-ellis — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml
Comment (real-rookie) [reply app/src/main/res/values/strings.xml:1]: thanks for the fix!

**Review by rachel-ellis — DISMISSED** [review]

**Review by nic3-14159 — APPROVED** [review]
lgtm

**Pull request merged**